librad/src/git/repo.rs

```diff
@@ -429,8 +435,8 @@ impl Repo {
     /// Obtain a [radicle-surf] `Browser` for this repo, consuming self.
     ///
     /// [radicle-surf]: https://github.com/radicle-dev/radicle-surf
-    pub fn browser(self) -> Result<surf::Browser, Error> {
-        let bro = surf::Browser::new(self.repo.into())?;
+    pub fn browser(&'_ self) -> Result<surf::Browser<'_>, Error> {
```

Comment:
This is correct now, but with #149 we won't be able to return a `Browser`. The reason is that we do not want to hand out a RAII lock to the library user -- while the lock is held, no progress can be made reacting to network events.

Instead, we want something like:

```rust
pub fn readonly(&self) -> Result<surf::Repository, Error> {
    let path = // ...
    surf::Repository::open(path, self.namespace())
}
```

Reply:
Depends on radicle-dev/radicle-surf#101, obviously

Reply:
Ah interesting :thinking: Can follow up with that after again once https://github.com/radicle-dev/radicle-surf/issues/101 is also in